perllib/Integrations/Bartec.pm

```diff
@@ -175,6 +175,46 @@ sub Premises_Get {
     return \@premises;
 }
 
+sub Premises_FutureWorkpacks_Get {
+    my ( $self, %args ) = @_;
+
+    my $res = $self->call(
+        'Premises_FutureWorkpacks_Get',
+        token    => $self->token,
+        UPRN     => $args{uprn},
+        DateFrom => $args{date_from},
+        DateTo   => $args{date_to},
+    );
+
+    # Already seem to be returned from Bartec in date ascending order, but
+    # we'll sort just in case
+    my $workpacks = force_arrayref( $res, 'Premises_FutureWorkPack' );
+    @$workpacks
+        = sort { $a->{WorkPackDate} cmp $b->{WorkPackDate} } @$workpacks;
+    return $workpacks;
+}
+
+sub WorkPacks_Get {
+    my ( $self, $date ) = @_;
+
+    my $res = $self->call(
+        'WorkPacks_Get',
+        token => $self->token,
+        Date  => {
+            MinimumDate => {
+                attr  => { xmlns => "http://www.bartec-systems.com" },
+                value => $date,
+            },
+            MaximumDate => {
+                attr  => { xmlns => "http://www.bartec-systems.com" },
+                value => $date,
```

Comment:
Although having the timestamp set to `00:00:00` for both `MinimumDate` and `MaximumDate` does appear to work, do you think it would be worth explicitly setting it to `23:59:59` for the latter?

Reply:
I'm not so sure about this; mainly because date munging of any kind can be such a faff, and it works as it currently stands.

Reply:
(As far as I have been able to tell.)

Reply:
This ought to do it :) 
```
my $start = $date->clone->set(hour => 0, minute => 0, second => 0);
my $end = $date->clone->set(hour => 23, minute => 59, second => 59);
```